### PR TITLE
make react agent prompts more editable

### PR DESCRIPTION
**Pull request by jerryjliu**
this is a bit more of a QoL change that allows users to more easily customize the react agent prompt.

- make ReActChatFormatter more easily importable
- make agent runners/workers import prompt mixin to easily view/update prompts
- remove duplicate react agent context notebook (consolidate it with core react query engine notebook) 
- improve intro example

### CODE REVIEW

File: llama_index/agent/react/formatter.py [hunk 1]
```diff
@@ -91,9 +91,31 @@ def format(
...
+
         return ReActChatFormatter(
-            context=context,
-            system_header=CONTEXT_REACT_CHAT_SYSTEM_HEADER,
+            system_header=system_header,
+            context=context or "",
+        )
+
+    @classmethod
+    def from_context(self, context: str) -> "ReActChatFormatter":
```
Comment (nerdai) [inline llama_index/agent/react/formatter.py:113]: nit: shouldn't this be `cls` instead of `self`?

File: llama_index/agent/react/formatter.py [hunk 2]
```diff
@@ -91,9 +91,31 @@ def format(
...
+        """Create ReActChatFormatter from context.
+
+        NOTE: deprecated
+
+        """
+        return ReActChatFormatter.from_defaults(
```
Comment (nerdai) [inline llama_index/agent/react/formatter.py:119]: should we raise a warning to indicate that this constructor method is deprecated?

File: llama_index/agent/types.py [hunk 3]
```diff
@@ -170,12 +170,25 @@ class Task(BaseModel):
...
+class BaseAgentWorker(PromptMixin):
     """Base agent worker."""
 
     class Config:
         arbitrary_types_allowed = True
 
+    def _get_prompts(self) -> PromptDictType:
+        """Get prompts."""
+        # TODO: the ReAct agent does not explicitly specify prompts, would need a
+        # refactor to expose those prompts
```
Comment (nerdai) [inline llama_index/agent/types.py:182]: Apologies for likely a dumb question: but do you mind explaining what you mean here?

**Review by nerdai — APPROVED** [review]
LGTM! Minor comments/questions.

File: docs/module_guides/deploying/agents/modules.md [hunk 4]
```diff
@@ -39,6 +39,7 @@ maxdepth: 1
 ---
 maxdepth: 1
 ---
+/examples/agent/react_agent.ipynb
 /examples/agent/react_agent_with_query_engine.ipynb
 /examples/agent/react_agent_with_initial_context.ipynb
```
Comment (hatianzhang) [inline docs/module_guides/deploying/agents/modules.md:44]: @jerryjliu should you remove this one
you are deleting it?

Comment (jerryjliu) [reply docs/module_guides/deploying/agents/modules.md:44]: good point

File: llama_index/agent/types.py (hunk 3 above)
Comment (jerryjliu) [reply llama_index/agent/types.py:182]: oops i'm just gonna remove

**Review by hatianzhang — APPROVED** [review]

**Pull request merged**